File: docs/articles_en/assets/snippets/compile_model_cpu.py
# Copyright (C) 2018-2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

from snippets import get_model


def main():
    model = get_model()

    #! [compile_model_default]
    import openvino as ov

    core = ov.Core()
    compiled_model = core.compile_model(model, "CPU")
    #! [compile_model_default]

    if "GPU" not in core.available_devices:
        return 0

    #! [compile_model_multi]
    core = ov.Core()
    compiled_model = core.compile_model(model, "MULTI:CPU,GPU.0")
    #! [compile_model_multi]

    #! [compile_model_auto]
    core = ov.Core()
    compiled_model = core.compile_model(model, "AUTO:CPU,GPU.0", {hints.performance_mode: hints.PerformanceMode.CUMULATIVE_THROUGHPUT})
    #! [compile_model_auto]

File: docs/articles_en/assets/snippets/compile_model_cpu.cpp
#include <openvino/runtime/core.hpp>


int main() {
    {
        //! [compile_model_default]
        ov::Core core;
        auto model = core.read_model("model.xml");
        auto compiled_model = core.compile_model(model, "CPU");
        //! [compile_model_default]
    }

    {
        //! [compile_model_multi]
        ov::Core core;
        auto model = core.read_model("model.xml");
        auto compiled_model = core.compile_model(model, "MULTI:CPU,GPU.0");
        //! [compile_model_multi]
    }

    {
        //! [compile_model_auto]
        ov::Core core;
        auto model = core.read_model("model.xml");
        auto compiled_model = core.compile_model(model, "AUTO:CPU,GPU.0", ov::hint::performance_mode(ov::hint::PerformanceMode::CUMULATIVE_THROUGHPUT));
        //! [compile_model_auto]
    }
}
